File: src/test/java/ev/projects/PlayerMother.java
package ev.projects;

public class PlayerMother {

    Player newPlayer() {
        return newPlayer(new Floor());
    }

    Player newPlayer(String name) {
        return new Player(new Board(), name);
    }

    Player newPlayer(Floor floor) {
        return newPlayer(new Wall(), floor);
    }

    Player newPlayer(Wall wall) {
        return newPlayer(wall, new Floor());
    }

    Player newPlayer(Wall wall, String name) {
        return new Player(new Board(wall, new Floor()), name);
    }

    Player newPlayer(Wall wall, Floor floor) {
        return new Player(new Board(wall, floor));
    }
}

File: python/lib/test/player_mother.py
from lib.azul.board import Board
from lib.azul.floor import Floor
from lib.azul.wall import Wall
from lib.azul.player import Player

class PlayerMother:
    @staticmethod
    def new_player(name=None, wall=None, floor=None):
        if floor is None:
            floor = Floor()
        if wall is None:
            wall = Wall()
        board = Board(wall=wall, floor=floor)
        if name:
            return Player(board, name)
        return Player(board)
